### DISCUSSION TITLE
Custom Style and Key cannot be null (BasedOn)

### DISCUSSION THREAD

**Discussion by Enat0** [HandyOrg/HandyControl#1244] [Q&A]
Hello

I have two themes (Dark and Light Default) and I also have 2 (Dark and Light) themes with additional styles I created.

Everything works fine and I don't have an error running the program, but I have several errors that I can't figure it out how to fix it.

For example I have this style:

```
<Style x:Key="ButtonNormal" TargetType="{x:Type Button}" BasedOn="{hc:StaticResource ButtonDefault}">
        <Setter Property="Background" Value="White"/>
        <Setter Property="Foreground" Value="#00256D"/>
        <Setter Property="BorderBrush" Value="#00256D"/>
        <Setter Property="BorderThickness" Value="1"/>
        <Style.Triggers>
            <Trigger Property="IsMouseOver" Value="True">
                <Setter Property="Background" Value="{DynamicResource SecondaryRegionBrush}"/>
                <Setter Property="Foreground" Value="{DynamicResource BorderBrushGeral}"/>
            </Trigger>
            <Trigger Property="IsPressed" Value="True">
                <Setter Property="Opacity" Value=".7"/>
                <Setter Property="Background" Value="#00256D"/>
                <Setter Property="BorderBrush" Value="#FF1491E8"/>
                <Setter Property="Foreground" Value="White"/>
            </Trigger>
        </Style.Triggers>
    </Style>
```

and Visual Studio 2022 is giving me this error on "BasedOn" : 

> 'Key cannot be null.
> Parameter name: key'

Again.. Everything is working fine and I have the custom style, but I also have that error in the error list..

**Comment by Yopler** [comment]
Hi 🖐,

You should remove the "hc:" before `StaticResource`.

**Reply by Enat0** [reply]
Hello,

That way I can't even run the program because I receive and error :s
